addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/handler/AbstractLedHandler.java

```diff
@@ -0,0 +1,360 @@
+                }
+                updateState(MilightBindingConstants.CHANNEL_COLOR, new HSBType(new DecimalType(state.hue360),
+                        new PercentType(state.saturation), new PercentType(state.brightness)));
+
+                break;
+            }
+            case MilightBindingConstants.CHANNEL_TEMP: {
+                if (command instanceof OnOffType) {
+                    OnOffType s = (OnOffType) command;
+                    this.setColorTemperature((s == OnOffType.ON) ? 100 : 0, state);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    this.changeColorTemperature(
+                            (IncreaseDecreaseType) command == IncreaseDecreaseType.INCREASE ? 1 : -1, state);
+                } else if (command instanceof DecimalType) {
+                    DecimalType d = (DecimalType) command;
+                    this.setColorTemperature(d.intValue(), state);
+                } else {
+                    logger.error("CHANNEL_TEMP channel only supports OnOffType/IncreaseDecreaseType/DecimalType");
+                }
+                break;
+            }
+            case MilightBindingConstants.CHANNEL_SPEED_REL: {
+                if (command instanceof IncreaseDecreaseType) {
+                    IncreaseDecreaseType id = (IncreaseDecreaseType) command;
+                    if (id == IncreaseDecreaseType.INCREASE) {
+                        this.changeSpeed(1, state);
+                    } else if (id == IncreaseDecreaseType.DECREASE) {
+                        this.changeSpeed(-1, state);
+                    }
+                } else {
+                    logger.error("CHANNEL_SPEED channel only supports IncreaseDecreaseType");
+                }
+                break;
+            }
+            case MilightBindingConstants.CHANNEL_ANIMATION_MODE: {
+                if (command instanceof DecimalType) {
+                    DecimalType d = (DecimalType) command;
+                    this.setLedMode(d.intValue(), state);
+                } else {
+                    logger.error("Animation mode channel only supports DecimalType");
+                }
+                break;
+            }
+            case MilightBindingConstants.CHANNEL_ANIMATION_MODE_REL: {
+                if (command instanceof IncreaseDecreaseType) {
+                    IncreaseDecreaseType id = (IncreaseDecreaseType) command;
+                    if (id == IncreaseDecreaseType.INCREASE) {
+                        this.nextAnimationMode(state);
+                    } else if (id == IncreaseDecreaseType.DECREASE) {
+                        this.previousAnimationMode(state);
+                    }
+                } else {
+                    logger.error("Relative animation mode channel only supports IncreaseDecreaseType");
+                }
+                break;
+            }
+            default:
+                logger.error("Channel unknown {}", channelUID.getId());
+        }
+    }
```

Comment:
That is a huge method?

Reply:
This was huge before, many other methods got smaller. I know that `AbstractLightHandler` deserves some slimming, but I don't feel like I wanna do it in this PR.